src/test/java/core/basesyntax/services/actions/PurchaseHandlerTest.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.services.actions;
+
+import core.basesyntax.db.Storage;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PurchaseHandlerTest {
+    private Map<String, ActionHandler> actionHandlerMap;
+
+    @Before
+    public void before() {
+        Storage.fruits.clear();
+        actionHandlerMap = new HashMap<>();
+        actionHandlerMap.put("b",new BalanceHandler());
+        actionHandlerMap.put("s",new IncreaseHandler());
+        actionHandlerMap.put("p",new PurchaseHandler());
+        actionHandlerMap.put("r",new IncreaseHandler());
+    }
+
+    @Test
+    public void purchaseHandler_Ok() {
+        String type = "p";
+        String fruitName = "apple";
+        int quantity = 20;
+        int expected = -20;
```

Comment:
we should not be able to store negative quantity in the storage
when the quantity is not enough to make a purchase - let's throw an exception?